src/analysis/explorerHomeView.js

```diff
@@ -0,0 +1,82 @@
+// @flow
+import {CredGraph} from "../core/credrank/credGraph";
+import {type TimestampMs} from "../util/timestamp";
+import {type Interval} from "../core/interval";
+import * as Grain from "../core/ledger/grain";
+
+export type DataOverTime = $ReadOnlyArray<number>;
+
+export type UserRow = {|
+  +name: string,
+  +credEarnedInTimeRange: number,
+  +grainHarvestedInTimeRange: Grain.Grain,
+  +grainBalance: Grain.Grain,
+  +allTimeCred: DataOverTime,
+  +category: UserCategory,
+|};
+
+export type UserCategory = "participant" | "bot" | "project" | "organization";
+
+export type ExplorerHomeView = {|
+  +credEarned: number,
+  +grainHarvested: Grain.Grain,
+  +participantCount: number, //Question: what is this supposed to capture? Is it supposed to change depending on the time range selected?
+  +grainPerCred: Grain.Grain,
+  +userRows: $ReadOnlyArray<UserRow>,
+  +intervalsIncluded: $ReadOnlyArray<Interval>,
+|};
+
+/**
+  Responsible for transforming data from the CredGraph and LedgerLog(?) into the information needed by the Explorer Home web page.
+ */
+export class ExplorerHomeViewBuilder {
+  constructor(credGraph: CredGraph) {
+    credGraph;
+    //Other params, depending on implementation. Probably a new GrainData API.
+    //Saves params or transforms them, depending on implementation strategy
+  }
+
+  /**
+    Returns all available intervals.
+    Useful for determining appropriate time ranges to offer for selection.
+   */
+  getAllIntervals(): $ReadOnlyArray<Interval> {
+    return [];
+  }
+
+  /**
+    Returns the timelines aggregated over all available intervals.
+   */
+  getAggregateTimelines(): {|
+    +credOverTime: DataOverTime,
+    +grainOverTime: DataOverTime,
```

Comment:
Grain is not a number, so this type signature is unsatisfiable

Reply:
it will need to be converted to a number to build line graphs. would you rather that happen downstream?

Reply:
Yeah, I think we should keep data accurate/fully faithful as long as possible. When we get to the line graphs, we can do numeric conversion locally as needed.